fineract-provider/src/main/java/org/apache/fineract/portfolio/delinquency/validator/DelinquencyRangeParseAndValidator.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.fineract.portfolio.delinquency.validator;
+
+import com.google.gson.JsonObject;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import javax.validation.constraints.NotNull;
+import lombok.RequiredArgsConstructor;
+import org.apache.fineract.infrastructure.core.api.JsonCommand;
+import org.apache.fineract.infrastructure.core.data.DataValidatorBuilder;
+import org.apache.fineract.infrastructure.core.serialization.FromJsonHelper;
+import org.apache.fineract.infrastructure.core.validator.ParseAndValidator;
+import org.apache.fineract.portfolio.delinquency.api.DelinquencyApiConstants;
+import org.apache.fineract.portfolio.delinquency.data.DelinquencyRangeData;
+import org.springframework.stereotype.Component;
+
+@RequiredArgsConstructor
+@Component
+public class DelinquencyRangeParseAndValidator extends ParseAndValidator {
+
+    private final FromJsonHelper jsonHelper;
+
+    public DelinquencyRangeData validateAndParseUpdate(@NotNull final JsonCommand command) {
```

Comment:
@josehernandezfintecheandomx Misleading method name. This validator is not used only for parsing and validating deliquency range update request.

Reply:
@josehernandezfintecheandomx I see no changes here... still has the same method name which refer to "Update" but this validation is used to validate and parse the "create" request as well. Wouldnt it be better to use "validateAndParseRequest"?